tokio/src/runtime/metrics/batch.rs

```diff
@@ -279,3 +287,12 @@ cfg_rt_multi_thread! {
 pub(crate) fn duration_as_u64(dur: Duration) -> u64 {
     u64::try_from(dur.as_nanos()).unwrap_or(u64::MAX)
 }
+
+/// `WASM` does not support time: <https://github.com/tokio-rs/tokio/issues/7319>
```

Comment:
The major problem here is that WASM does support time ... sometimes. Perhaps we could change this logic to only apply when you enable the `time` feature of Tokio? I'm not sure if that would be breaking.

Reply:
That seems reasonable. In the longer term, we could add a builder option to opt in or out of these time based metrics.